iohub/reader.py

```diff
@@ -65,75 +41,57 @@ def _check_zarr_data_type(src: str):
     return "unknown"
 
 
-def _check_single_page_tiff(src: str):
-    # pick parent directory in case a .tif file is selected
-    if src.endswith(".tif"):
-        src = os.path.dirname(src)
-
-    files = glob.glob(os.path.join(src, "*.tif"))
+def _check_single_page_tiff(src: Path):
+    if src.is_file():
+        src = src.parent
+    files = src.glob("*.tif")
     if len(files) == 0:
```

Comment:
Could you check if `len(files)` work?
I think it will return ```TypeError:` object of type 'generator' has no len()```

Reply:
Thanks for catching this! Also added a new test.